components/audioinput/microphone/microphone.go

```diff
@@ -77,30 +77,38 @@ func newMicrophoneSource(attrs *Attrs, logger golog.Logger) (audioinput.AudioInp
 			return nil, err
 		}
 	}
-
-	labels := gostream.QueryAudioDeviceLabels()
-	for _, label := range labels {
-		if debug {
-			logger.Debugf("%s", label)
-		}
-
-		if pattern != nil && !pattern.MatchString(label) {
-			if debug {
-				logger.Debug("\t skipping because of pattern")
+	all := gostream.QueryAudioDevices()
+
+	for i, info := range all {
+		logger.Debugf("%s", info.ID)
+		logger.Debugf("\t labels: %v", info.Labels)
+		for _, p := range info.Properties {
+			logger.Debugf("\t %+v", p.Audio)
+			if pattern != nil && !pattern.MatchString(info.Labels[i]) {
+				if debug {
+					logger.Debug("\t skipping because of pattern")
+				}
+				continue
 			}
```

Comment:
You can move this check up a level: Since this check is only using the `i` variable, you can have it continue faster if it doesn't match the label, rather than checking on all the p loops.

Reply:
done